scripts/patches/json.diff

```diff
@@ -1,62 +1,64 @@
+@@ -23,6 +23,9 @@ import com.google.gson.JsonElement;
  import io.gsonfire.GsonFireBuilder;
  import io.gsonfire.TypeSelector;
  
++import io.kubernetes.client.gson.V1MetadataExclusionStrategy;
 +import io.kubernetes.client.gson.V1StatusPreProcessor;
 +import io.kubernetes.client.openapi.models.V1Status;
  import okio.ByteString;
  
  import java.io.IOException;
-@@ -34,6 +36,9 @@ import java.text.ParsePosition;
- import java.time.LocalDate;
- import java.time.OffsetDateTime;
+@@ -35,6 +38,9 @@ import java.time.OffsetDateTime;
+ import java.time.ZoneId;
+ import java.time.ZoneOffset;
  import java.time.format.DateTimeFormatter;
 +import java.time.format.DateTimeFormatterBuilder;
 +import java.time.format.DateTimeParseException;
 +import java.time.temporal.ChronoField;
  import java.util.Date;
  import java.util.Locale;
  import java.util.Map;
-@@ -48,9 +53,19 @@ import java.util.HashMap;
+@@ -50,9 +56,20 @@ import java.util.TimeZone;
  public class JSON {
      private static Gson gson;
      private static boolean isLenientOnJson = false;
 +
 +    private static final DateTimeFormatter RFC3339MICRO_FORMATTER =
-+            new DateTimeFormatterBuilder()
-+                    .parseDefaulting(ChronoField.OFFSET_SECONDS, 0)
-+                    .append(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss"))
-+                    .optionalStart()
-+                    .appendFraction(ChronoField.NANO_OF_SECOND, 6, 6, true)
-+                    .optionalEnd()
-+                    .appendLiteral("Z")
-+                    .toFormatter();
++        new DateTimeFormatterBuilder()
++            .parseDefaulting(ChronoField.OFFSET_SECONDS, 0)
++            .append(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss"))
++            .optionalStart()
++            .appendFraction(ChronoField.NANO_OF_SECOND, 6, 6, true)
++            .optionalEnd()
++            .appendLiteral("Z")
++            .toFormatter();
++
      private static DateTypeAdapter dateTypeAdapter = new DateTypeAdapter();
      private static SqlDateTypeAdapter sqlDateTypeAdapter = new SqlDateTypeAdapter();
 -    private static OffsetDateTimeTypeAdapter offsetDateTimeTypeAdapter = new OffsetDateTimeTypeAdapter();
 +    private static OffsetDateTimeTypeAdapter offsetDateTimeTypeAdapter = new OffsetDateTimeTypeAdapter(RFC3339MICRO_FORMATTER);
      private static LocalDateTypeAdapter localDateTypeAdapter = new LocalDateTypeAdapter();
      private static ByteArrayAdapter byteArrayAdapter = new ByteArrayAdapter();
  
-@@ -58,7 +73,10 @@ public class JSON {
+@@ -65,8 +82,11 @@ public class JSON {
      public static GsonBuilder createGson() {
          GsonFireBuilder fireBuilder = new GsonFireBuilder()
          ;
 -        GsonBuilder builder = fireBuilder.createGsonBuilder();
+-        return builder;
 +        GsonBuilder builder =
 +                fireBuilder
 +                        .registerPreProcessor(V1Status.class, new V1StatusPreProcessor())
```

Comment:
@brendandburns looks like the patch has been broken for a while, curiously did you change JSON.java manually when generating 1.30?

Reply:
Yeah, I think I broke this when I regenerated 1.30